tasks/verifyOffsetHelper.ts

```diff
@@ -0,0 +1,28 @@
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import addresses, { mumbaiAddresses } from "../utils/addresses";
+
+task("verify:offsetHelper", "Verifies the OffsetHelper")
+  .addParam("address", "The OffsetHelper address")
+  .setAction(
+    async (taskArgs: { address: any }, hre: HardhatRuntimeEnvironment) => {
+      const { address } = taskArgs;
+
+      const addressesToUse =
+        hre.network.name == "mumbai" ? mumbaiAddresses : addresses;
+
+      await hre.run("verify:verify", {
```

Comment:
I know this is how `tokenizer` does it for TCO2s, but that doesn't necessarily make it right in other scenarios... or even at all ;-)

Worth remembering that TCO2s are a special case since they are created dynamically from a factory.

Also, I'm wondering if https://www.npmjs.com/package/hardhat-deploy#user-content-4-hardhat-etherscan-verify is maybe a cleaner approach here.  No strong opinions, but would be cool if you could look into that - or probably @0xmichalis already knows the pros and cons :-)

Reply:
Agreed with Adam, I am not sure this task adds much value here as it reimplements `hardhat verify` and it also expects to have the addresses in `utils` updated (so it's not as flexible as using `hardhat verify`).

Reply:
The value (to me) of this task is that I do not have to copy over the addresses every single time I want to verify the contract. Having to write `["BCT", "NCT", ...] ["0x123", "0x321", ...]` is something I don't really want to do.

As for the:
> it also expects to have the addresses in utils updated 

Deploying runs using these addresses too, so if we're going to make an issue of it in verification we should make an issue of it in deployment as well.

Reply:
Talked to Michalis on google meet. Note to self: document native `hardhat verify` in README as well as the custom task and we should be fine.